app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.services
+
+import javax.inject.{Inject, Singleton}
+import play.api.Logger
+import play.api.mvc.Result
+import play.api.mvc.Results._
+import uk.gov.hmrc.exports.metrics.ExportsMetrics
+import uk.gov.hmrc.exports.metrics.MetricIdentifiers.notificationMetric
+import uk.gov.hmrc.exports.models.DeclarationNotification
+import uk.gov.hmrc.exports.repositories.{NotificationsRepository, SubmissionRepository}
+import uk.gov.hmrc.wco.dec.Response
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class NotificationService @Inject()(
+  submissionRepository: SubmissionRepository,
+  notificationsRepository: NotificationsRepository,
+  metrics: ExportsMetrics
+) {
+
+  private val logger = Logger(this.getClass)
+
+  def save(notification: DeclarationNotification): Future[Result] = {
+    val eori = notification.eori
+    val convId = notification.conversationId
+
+    for {
+      oldNotification <- notificationsRepository
+        .getByEoriAndConversationId(eori, convId)
+        .map(_.sortWith((a, b) => a.isOlderThan(b)).headOption)
```

Comment:
I am not entirely sure, but isn't it that now we will have the oldest notification at the head? I thought we want to get the "youngest" notification stored so far.

Reply:
This method is covered by tests.

`DeclarationNotificationSpec`. I'll check that it covers all logic, but the method should works correctly

Reply:
added filtering by date and improved `isOlderThan` method